src/mpid/ch4/src/ch4_coll.h

```diff
@@ -1362,4 +1362,18 @@ MPL_STATIC_INLINE_PREFIX int MPID_Iscatterv(const void *sendbuf, const int *send
     return ret;
 }
 
+MPL_STATIC_INLINE_PREFIX int MPID_Bcast_init(void *buffer, int count, MPI_Datatype datatype,
+                                             int root, MPIR_Comm * comm_ptr, MPIR_Info * info_ptr,
+                                             MPIR_Request ** request)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPID_BCAST_INIT);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPID_BCAST_INIT);
+
+    mpi_errno = MPIR_Bcast_init(buffer, count, datatype, root, comm_ptr, info_ptr, request);
```

Comment:
Shouldn't it be `MPIR_Bcast_init_impl`? `MPIR_Bcast_init` will call back into `MPID_Bcast_init`, right? Is there some mechanism preventing this circular call that I am not aware of?

Reply:
I think right now it works because it goes as `MPIX_Bcast_init` -> `MPID_Bcast_init` -> `MPIR_Bcast_init`. 
Would we still need `MPIX`? I think `X` was there because it wasn't a part of the standard when we added it.